src/nrc/api/tasks.py

```diff
@@ -35,16 +49,23 @@ def deliver_message(sub_id: int, msg: dict, **kwargs) -> None:
             headers={"Content-Type": "application/json", "Authorization": sub.auth},
         )
         response_init_kwargs = {"response_status": response.status_code}
+        if response.status_code != 204:
+            exception_message = _(
+                "Could not send notification: status {status_code} - {response}"
+            ).format(status_code=response.status_code, response=response.text)
+            response_init_kwargs["exception"] = exception_message
+            raise NotificationException(exception_message)
     except requests.exceptions.RequestException as e:
-        # log of the response of the call
         response_init_kwargs = {"exception": str(e)}
-
-    logger.debug(
-        "Notification response for %d, %r: %r", sub_id, msg, response_init_kwargs
-    )
-
-    # Only log if a top-level object is provided
-    if notificatie_id:
-        NotificatieResponse.objects.create(
-            notificatie_id=notificatie_id, abonnement=sub, **response_init_kwargs
+        raise e
```

Comment:
```suggestion
        raise
```
simple `raise` will re-raise the original exception